### PR TITLE
Fix all racy calls to setState

### PR DESCRIPTION
**Pull request by akalin-keybase**
replaced setState( .. this.state .. ) with
setState(prevState => .. prevState .. ).

### CODE REVIEW

File: shared/chat/conversation/list/index.desktop.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ class BaseList extends React.Component<Props, State> {
   componentDidUpdate(prevProps: Props, prevState: State) {
     // Force a rerender if we passed a row to scroll to. If it's kept around the virutal list gets confused so we only want it to render once basically
     if (this._keepIdxVisible !== -1) {
-      this.setState({listRerender: this.state.listRerender + 1}) // eslint-disable-line react/no-did-update-set-state
+      this.setState({listRerender: prevState.listRerender + 1}) // eslint-disable-line react/no-did-update-set-state
```
Comment (chrisnojima) [inline shared/chat/conversation/list/index.desktop.js]: needs to be a closure

Comment (akalin-keybase) [reply shared/chat/conversation/list/index.desktop.js]: You're right! Fixed.

**Comment by akalin-keybase** [comment]
PTAL

**Pull request merged**